### PR TITLE
feat: added direct calls to native precompiles #647

### LINKED ISSUE #647: Direct Calls to Native Precompiles Tests
### Problem

We recently enabled direct calls to ethereum precompiles, but don't have e2e tests for this.

### Solution

Add tests which cover this functionality.

- [ ] Call to 0x01 - ecRecover
- [ ] Call to 0x02 - SHA2 - 256
- [ ] Call to 0x03 - RIPEMD-160
- [ ] Call to 0x04 - identity
- [ ] Call to 0x05 - modexp
- [ ] Call to 0x06 - ecAdd
- [ ] Call to 0x07 - ecMul
- [ ] Call to 0x08 - ecPairing
- [ ] Call to 0x09 - blake2f

### Prerequisites
 Test against the `latest` local node version. 
(and make sure the mirror node version is:  **0.98.0-SNAPSHOT**)

Precompile cheat sheet: 
https://www.evm.codes/precompiled?fork=shanghai

_No response_

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Problem
We recently enabled direct calls to ethereum precompiles, but don't have e2e tests for this.

Solution
Add tests which cover this functionality.

Direct Calls to Native Precompile Tests
fixes #647

**Related issue(s)**:
[#647](https://github.com/hashgraph/hedera-smart-contracts/issues/647)
it's a subtask of:
[#7568](https://github.com/hashgraph/hedera-mirror-node/issues/7568)

### CODE REVIEW

**Comment by quiet-node** [comment]
Hi @stefan-stefanooov, could you please update the PR for it to be more informative. Also, I believe that `fixes #647` is necessary in the description for the associated issue to be closed after this PR got merged.

**Review by quiet-node — CHANGES_REQUESTED** [review]
LGTM. Just 1 request please

File: artifacts/contracts/exchange-rate-precompile/ExchangeRateMock.sol/ExchangeRateMock.json [hunk 1]
```diff
@@ -68,8 +68,8 @@
...
...7f019c356cd1ab2a7445bef30fa8b9e32be6636e6efae70244c7b3e0816040516100b191815260200190565b60405180910390a1919050565b60006100c98261020d565b90507fc399bb30ec076dee6179663a550714898c27a4a8cf23a336b2e49ec777779459816040516100b191815260200190565b600080600061016873ffffffffffffffffffffffffffffffffffffffff16632e3cff6a60e01b8560405160240161013591815260200190565b60408051601f198184030181529181526020820180517bffffffffffffffffffffffffffffffffffffffffffffffffffffffff167fffffffff000000000000000000000000000000000000000000000000000000009094169390931790925290516101a0919061025f565b6000604051808303816000865af19150503d80600081146101dd576040519150601f19603f3d011682016040523d82523d6000602084013e6101e2565b606091505b5091509150816101f157600080fd5b80806020019051810190610205919061028e565b949350505050565b600080600061016873ffffffffffffffffffffffffffffffffffffffff166343a8822960e01b8560405160240161013591815260200190565b60006020828403121561025857600080fd5b5035919050565b6000825160005b818110156102805760208186018101518583015201610266565b506000920191825250919050565b6000602082840312156102a057600080fd5b505191905056fea2646970667358221220495bc7cad00404d53b11268c14618130cc20e8c19c1f9c135f0ecd267d698d4a64736f6c63430008140033",
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply artifacts/contracts/exchange-rate-precompile/ExchangeRateMock.sol/ExchangeRateMock.json]: Rebased ! Thank you @quiet-node

File: test/eth-precompile/EthPrecompiles.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
+/*-
```
Comment (quiet-node) [inline test/eth-precompile/EthPrecompiles.js:1]: Just for the consistency, can you please move this file into the native-precompiles/ folder? Just how you had the EthNativePrecompileCaller.sol in the contract/native-precompiles

**Review by quiet-node — DISMISSED** [review]
LGTM. Thanks for the work!

**Review by quiet-node — APPROVED** [review]
LGTM

**Pull request merged**